checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,293 @@
+      LOG.info("Spark driver can find: " + alluxioURIClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find AlluxioURI class:" + e);
+      return 1;
+    }
+
+    try {
+      Class alluxioBaseFSClass = Class.forName("alluxio.client.file.BaseFileSystem");
+      LOG.info("Spark driver can find: " + alluxioBaseFSClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find alluxio.client.file.BaseFileSystem class:" + e);
+      return 1;
+    }
+
+    try {
+      Class alluxioHadoopFSClass = Class.forName("alluxio.hadoop.AlluxioFileSystem");
+      LOG.info("Spark driver can find: " + alluxioHadoopFSClass.getName());
+    } catch (ClassNotFoundException e) {
+      LOG.error("Spark driver cannot find alluxio.hadoop.AlluxioFileSystem class:" + e);
+      return 1;
+    }
+
+    // check if Spark driver can recognize Alluxio filesystem
+    try {
+      Class alluxioFSClass = FileSystem.getFileSystemClass("alluxio", new Configuration());
+      LOG.info("Spark driver can recognize Alluxio filesystem: " + alluxioFSClass.getName());
+    } catch (Exception e) {
+      LOG.error("Spark driver cannot recognize Alluxio filesystem: " + e);
+      return 2;
+    }
+
+    // support Alluxio high availability mode
+    if (alluxio.Configuration.getBoolean(PropertyKey.ZOOKEEPER_ENABLED)) {
+      sAlluxioHAMode = true;
+      try {
+        sZookeeperAddress = alluxio.Configuration.get(PropertyKey.ZOOKEEPER_ADDRESS);
+      } catch (RuntimeException e) {
+        LOG.error("Zookeeper address has not been set:" + e);
+        return 3;
+      }
+    }
+
+    // run a Spark job to check Spark executors configuration
+    return runOperations(sc);
+  }
+
+  /**
+   * Spark job to check whether Spark executors can recognize Alluxio filesystem.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 on success, 1 on failures
+   */
+  private static int runOperations(JavaSparkContext sc) {
+    // generate a list of integer for testing
+    List<Integer> nums = IntStream.rangeClosed(1, 1000).boxed().collect(Collectors.toList());
+
+    //TODO(lu) set a reasonable parition number to check as many executors as possible
+    JavaRDD<Integer> dataSet = sc.parallelize(nums);
+
+    // run a Spark job to check whether Spark executors can recognize Alluxio
+    JavaPairRDD<Integer, String> extractedStatus = dataSet
+        .mapToPair(s -> new Tuple2<>(getStatus(s), getAddress(s)));
+
+    // merge the IP addresses that can/cannot recognize Alluxio
+    JavaPairRDD<Integer, String> mergeStatus = extractedStatus.reduceByKey((a, b) -> merge(a, b));
+
+    sSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong happen
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Integer, String> op : sSparkJobResult) {
+      if (op._1() == 1) {
+        canRecognizeClass = false;
```

Comment:
you can return 1 here, and return 2 at line 137. Then at line 140 you can return 0

Reply:
Thanks for suggestions!
Unable to recognize Alluxio FileSystem maybe because they cannot recognize Alluxio classes, so if both failures happened, I hope to print helping message of the more fundamental one(fail to recognize Alluxio classes).

Reply:
then you can swap the two condition checks?